### ISSUE TITLE
print in JS sometimes prints array of char instead of string

### ISSUE THREAD

**Issue by samuelgruetter** [dafny-lang/dafny#258]
```
method Main() {
    var a := new char[2];
    a[0] := 'h';
    a[1] := 'i';
    print a[..];
}
```

When I compile this to C# and run it, it outputs `hi`, as expected.
However, when I compile it to JS and run it with `node`, it outputs `[h, i]`.
The behavior should be the same, no matter which language I export to.

(considered low priority, just posting here for reference)

**Comment by davidcok** [comment]
cok-258 has a test and fix for java. Not yet for go and js.

**Comment by davidcok** [comment]
Fixed by PR #675. One remaining infelicity: when Dafny sequences are passed through a generic interface as in method pr<T>(s: seq<T>), then JS and GO cannot tell the element type of an empty string, so empty sequences passed through such an interface will print as [] instead of just nothing, as C# and Java will do.

**Comment by robin-aws** [comment]
> Fixed by PR #675. One remaining infelicity: when Dafny sequences are passed through a generic interface as in method pr(s: seq), then JS and GO cannot tell the element type of an empty string, so empty sequences passed through such an interface will print as [] instead of just nothing, as C# and Java will do.

@RustanLeino - note this is a reason to keep the RTTI values in Java sequences, or some equivalent way of identifying `seq<char>` values at runtime. I updated #413 with the plans we discussed, just to emphasize the fact that I don't think we'll be changing things to make strings a separate type (which would be another way to solve this).
